File: desktop/src/com/inkostilation/pong/desktop/network/Network.java
package com.inkostilation.pong.desktop.network;

import com.inkostilation.pong.engine.IEngine;

public class Network {

    private static IEngine engine = null;

    public static IEngine getEngine() {
        if (engine == null) {
            engine = new ServerEngine();
        }
        return engine;
    }

}

File: desktop/src/com/inkostilation/pong/desktop/network/Network.kt
package com.inkostilation.pong.desktop.network

import com.inkostilation.pong.engine.IEngine

object Network {
    var engine: IEngine<*>? = null
        get() {
            if (field == null) {
                field = NetworkConnector()
            }
            return field
        }
        private set
}